library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/impl/ModResourcePackUtil.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.impl;
+
+import java.io.InputStream;
+
+import com.google.common.base.Charsets;
+import net.fabricmc.loader.api.metadata.ModMetadata;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.SharedConstants;
+import net.minecraft.resource.ResourceType;
+
+@ApiStatus.Internal
+public final class ModResourcePackUtil {
+	public static boolean containsDefault(ModMetadata info, String filename) {
+		return "pack.mcmeta".equals(filename);
+	}
+
+	public static InputStream openDefault(ModMetadata info, ResourceType type, String filename) {
+		if ("pack.mcmeta".equals(filename)) {
+			String description = info.getName();
+
+			if (description == null) {
+				description = "";
+			} else {
+				description = description.replaceAll("\"", "\\\"");
+			}
+
+			var pack = String.format("""
```

Comment:
This formatting is a bit bizzare?

Reply:
How should this be formatted then?
A text block expression requires the string to begin on a new line.